### PR TITLE
Martin/ios 741 player restart after buffer

### PR DESCRIPTION
**Pull request by MartinJibber**
This fixes issues with videos no longer playing and also fixes issues where the still images stop appearing after cells are recycled.

### CODE REVIEW

File: Shared/UI/Views/ExpressionVideoView.swift [hunk 1]
```diff
@@ -12,24 +12,32 @@ class ExpressionVideoView: VideoView {
 
     var expression: Expression? {
         didSet {
+            // Only update the video if this is a new expression.
+            guard self.expression != oldValue else { return }
```
Comment (MartinJibber) [inline Shared/UI/Views/ExpressionVideoView.swift:16]: Optimization.

File: Shared/UI/Views/ExpressionVideoView.swift [hunk 2]
```diff
@@ -12,24 +12,32 @@ class ExpressionVideoView: VideoView {
...
+
+            self.videoURL = nil
```
Comment (MartinJibber) [inline Shared/UI/Views/ExpressionVideoView.swift:18]: Hide the previous video while loading the new video.

File: Shared/UI/Views/ExpressionVideoView.swift [hunk 3]
```diff
@@ -12,24 +12,32 @@ class ExpressionVideoView: VideoView {
...
+    /// The currently running task that loads the video url.
     private var loadTask: Task<Void, Never>?
     
     private func updatePlayer(with expression: Expression?) {
         self.loadTask?.cancel()
 
+        guard let expression = self.expression else {
+            self.videoURL = nil
+            return
+        }
```
Comment (MartinJibber) [inline Shared/UI/Views/ExpressionVideoView.swift:32]: If no expression is set, hide the previous video.

File: Shared/UI/Views/MessageContentView/MessageContentView.swift [hunk 4]
```diff
@@ -314,21 +314,24 @@ class MessageContentView: BaseView {
         self.loadTask?.cancel()
         
         self.loadTask = Task { [weak self] in
-            guard let `self` = self else { return }
```
Comment (MartinJibber) [inline Shared/UI/Views/MessageContentView/MessageContentView.swift:317]: Removing this. This guard will create a strong reference throughout the life of the task. We don't really want that if the task is view removed before the task completes.

File: Shared/UI/Views/VideoView.swift [hunk 5]
```diff
@@ -35,50 +35,6 @@ class VideoView: BaseView {
...
-                        break
-                    }
-                case .failed:
-                    break
-                @unknown default:
-                    break
-                }
-                
-            }.store(in: &self.cancellables)
-
```
Comment (MartinJibber) [inline Shared/UI/Views/VideoView.swift:81]: I don't think we need this? Knowing about it is very helpful though.

File: Shared/UI/Views/VideoView.swift [hunk 6]
```diff
@@ -88,23 +44,35 @@ class VideoView: BaseView {
...
+
         guard let videoURL = url else {
+            // Stop playback if the url is nil.
             self.playerLayer.player = nil
             return
         }
-        
-        Task {
+
+        self.loadTracksTask = Task {
```
Comment (MartinJibber) [inline Shared/UI/Views/VideoView.swift:59]: Making this task cancellable as well.

File: Shared/UI/Views/VideoView.swift (hunk 5 above)
Comment (Benjamindodgson) [reply Shared/UI/Views/VideoView.swift:81]: Yeah I was just using if for testing. Fine to remove it.

**Review by Benjamindodgson — APPROVED** [review]

**Pull request merged**